core-6/src/main/java/configuration/BeanConfiguration.java

```diff
@@ -0,0 +1,67 @@
+package configuration;
+
+import configuration.car.carImpl.*;
+import configuration.engine.Engine;
+import configuration.engine.engineImpl.FourCylinderEngine;
+import configuration.engine.engineImpl.V12Engine;
+import configuration.engine.engineImpl.V6Engine;
+import configuration.parkinglot.parkinglotImpl.FreeParkingLot;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.*;
+import org.springframework.lang.Nullable;
+
+@Configuration
+@PropertySource(value="context.properties")
+public class BeanConfiguration {
+    @Bean
+    TeslaHoverPlus teslaHoverPlus(@Qualifier("plasmaEngine")@Nullable Engine engine){
```

Comment:
@Nullable and @Autowired(required = false) are the same, let's make it differently